control/iosys.py

```diff
@@ -908,6 +909,7 @@ def __init__(self, syslist, connections=[], inplist=[], outlist=[],
             self.state_offset.append(nstates)
 
             # Keep track of the total number of states, inputs, outputs
+            nstatenames += [sys.name + statename for statename in list(sys.state_index)]
```

Comment:
I'm thinking about fixing the <None> name problem at a higher level in the class hierarchy.
I would add a default name "sys[n]" (where n would be an incrementing counter) for every unnamed InputOutputSystem. This would change certain default outputs however, such as line 169:
`str = "System: " + (self.name if self.name else "(None)") + "\n"`
What do you think? Can I go ahead with such changes?

Reply:
Seems fine to assign default names if none is specified.   I suggest using 'x' for state, 'u' for input, 'y' for output, since this is the common terminology used in textbooks.

Reply:
This is already what is used. Default names are x[i], u[i], y[i]. So I'm thinking of using sys[i] for default sys name (instead of having None).
When we interconnect systems, I would call default state names sys[i].x[j]
This is because it always annoyed me in my code that when I cascaded a PID with a plant, the state names would revert back to the defaults x[0],x[1],x[2],etc. and I would lose track of which state belongs to which subsystem.
Is this ok?

Reply:
Oops, sorry, I misunderstood.  Seems reasonable.

The system and signal names was done in a hurry.  It would be great to have a better implementation.

Don't forget to add unit tests (including situations that should fail and make sure that they do with the appropriate error message) and also update the documentation.